catalog/glue.go

```diff
@@ -0,0 +1,186 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package catalog
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/apache/iceberg-go/io"
+	"github.com/apache/iceberg-go/table"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/glue"
+	"github.com/aws/aws-sdk-go-v2/service/glue/types"
+)
+
+const glueTableTypeIceberg = "ICEBERG"
+
+var (
+	_ Catalog = (*GlueCatalog)(nil)
+)
+
+type glueAPI interface {
+	GetTable(ctx context.Context, params *glue.GetTableInput, optFns ...func(*glue.Options)) (*glue.GetTableOutput, error)
+	GetTables(ctx context.Context, params *glue.GetTablesInput, optFns ...func(*glue.Options)) (*glue.GetTablesOutput, error)
```

Comment:
I think `LoadTable` is clearer, as I generally try to avoid `Get` in the name. Why do we need `GetTables`? Wouldn't calling LoadTable multiple times give the same result? This also ties in with the comment on `ErrNoSuchTable` above; if we don't give the table name in the exception, you don't know which one doesn't exists.

Reply:
I think this is describing the AWS Glue API, which has `GetTable` and `GetTables` API methods

Reply:
@Fokko yeah @jackye1995 is 100% correct on this, I am using an interface to enable mocking of just the operations in the AWS I want to consume. This pattern is covered here and is widely used in Go.  https://aws.github.io/aws-sdk-go-v2/docs/unit-testing/#mocking-client-operations

Reply:
One thing to note is this interface is private as well thanks to feedback from @zeroshade .